RecoEgamma/EgammaTools/src/EffectiveAreas.cc

```diff
@@ -0,0 +1,119 @@
+#include "RecoEgamma/EgammaTools/interface/EffectiveAreas.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include <math.h>
+#include <fstream>
+#include <string>
+#include <sstream>
+
+EffectiveAreas::EffectiveAreas(TString filename):
+  filename_(filename)
+{
+
+  // Open the file with the effective area constants
+  std::ifstream inputFile;
+  inputFile.open(filename_.Data());
+  if( !inputFile.is_open() )
+    throw cms::Exception("EffectiveAreas config failure")
+      << "failed to open the file " << filename_.Data() << std::endl;
+  
+  // Read file line by line
+  std::string line;
+  const float undef = -999;
+  while( getline(inputFile, line) ){
+    if(line[0]=='#') continue; // skip the comments lines
+    float etaMin = undef, etaMax = undef, effArea = undef;
+    std::stringstream ss(line);
+    ss >>  etaMin >> etaMax >> effArea;
+    // In case if the format is messed up, there are letters
+    // instead of numbers, or not exactly three numbers in the line,
+    // it is likely that one or more of these vars never changed
+    // the original "undef" value:
+    if( etaMin==undef || etaMax==undef || effArea==undef )
+      throw cms::Exception("EffectiveAreas config failure")
+	<< "wrong file format, file name " << filename_.Data() << std::endl;
+    
+    absEtaMin_          .push_back( etaMin );
+    absEtaMax_          .push_back( etaMax );
+    effectiveAreaValues_.push_back( effArea );
+  }
+
+  // Extra consistency checks are in the function below.
+  // If any of them fail, an exception is thrown.
+  checkConsistency();
+}
+
+EffectiveAreas::~EffectiveAreas(){
+
+  absEtaMin_.clear();
+  absEtaMax_.clear();
+  effectiveAreaValues_.clear();
+
+}
+
+// Return effective area for given eta
+const float EffectiveAreas::getEffectiveArea(float eta) const{
+
+  float effArea = 0;
+  uint nEtaBins = absEtaMin_.size();
+  for(uint iEta = 0; iEta<nEtaBins; iEta++){
+    if( fabs(eta) >= absEtaMin_.at(iEta)
+	&& fabs(eta) < absEtaMax_.at(iEta) ){
+      effArea = effectiveAreaValues_.at(iEta);
+      break;
+    }
+  }
+
+  return effArea;
+}
+
+void EffectiveAreas::printEffectiveAreas(){
+
+  printf("EffectiveAreas: source file %s\n", filename_.Data());
```

Comment:
CMS coding rules say not to use printf.
This function is currently not used though.